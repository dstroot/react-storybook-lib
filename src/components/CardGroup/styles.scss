/* SVG Stuff */

.svg-container {
  width: 30%;
  padding-top: 5%;
  fill: theme-color("primary");
}
