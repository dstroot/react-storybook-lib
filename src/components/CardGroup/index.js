import React from "react";

// styles
import "./styles.scss";

// import SVG files
import { ReactComponent as Flag } from "./media/flag.svg";
import { ReactComponent as Locked } from "./media/locked.svg";
import { ReactComponent as Handshake } from "./media/handshake.svg";

const CardGroup = () => {
  const textBig = {
    fontSize: `calc(26px + 1.5vw + 0.5vmin)`,
    lineHeight: "1.2",
    fontWeight: "300"
  };

  return (
    <div className="container-fluid bg-primary text-center text-white">
      <div className="row">
        <div className="col-10 offset-1">
          <p style={textBig} className="pt-5">
            What the Power of Pacific Means for You
          </p>
          <div className="card-group my-3">
            <div className="card bg-primary border-0">
              <div className="svg-container mx-auto">
                <Flag />
              </div>
              <div className="card-body">
                <h3 className="card-title">Freedom</h3>
                <p className="card-text">
                  Life insurance and retirement solutions to help you reach your
                  goals.
                </p>
              </div>
            </div>
            <div className="card bg-primary border-0">
              <div className="svg-container mx-auto">
                <Locked />
              </div>
              <div className="card-body">
                <h3 className="card-title">Security</h3>
                <p className="card-text">
                  A deep commitment to protecting generations of families.
                </p>
              </div>
            </div>
            <div className="card bg-primary border-0">
              <div className="svg-container d-block mx-auto align-bottom">
                <Handshake />
              </div>
              <div className="card-body">
                <h3 className="card-title">Trust</h3>
                <p className="card-text">
                  150 years of strength and stability by your side.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CardGroup;
