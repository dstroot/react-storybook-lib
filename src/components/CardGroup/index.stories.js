import React from "react";
import { storiesOf } from "@storybook/react";

// components
import CardGroup from "../CardGroup";

// CardGroup section
storiesOf("CardGroup", module)
  .addParameters({
    info: {
      inline: true,
      header: false
    }
  })
  .add("show CardGroup", () => <CardGroup />);
