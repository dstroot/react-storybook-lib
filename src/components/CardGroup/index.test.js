import React from 'react';
import renderer from 'react-test-renderer';
import CardGroup from '../CardGroup';

describe('CardGroup', () => {
  it('it should render', () => {
    const component = renderer.create(<CardGroup />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
