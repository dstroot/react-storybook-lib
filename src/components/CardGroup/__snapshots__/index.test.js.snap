// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGroup it should render 1`] = `
<div
  className="container-fluid bg-primary text-center text-white"
>
  <div
    className="row"
  >
    <div
      className="col-10 offset-1"
    >
      <p
        className="pt-5"
        style={
          Object {
            "fontSize": "calc(26px + 1.5vw + 0.5vmin)",
            "fontWeight": "300",
            "lineHeight": "1.2",
          }
        }
      >
        What the Power of Pacific Means for You
      </p>
      <div
        className="card-group my-3"
      >
        <div
          className="card bg-primary border-0"
        >
          <div
            className="svg-container mx-auto"
          >
            <svg>
              flag.svg
            </svg>
          </div>
          <div
            className="card-body"
          >
            <h3
              className="card-title"
            >
              Freedom
            </h3>
            <p
              className="card-text"
            >
              Life insurance and retirement solutions to help you reach your goals.
            </p>
          </div>
        </div>
        <div
          className="card bg-primary border-0"
        >
          <div
            className="svg-container mx-auto"
          >
            <svg>
              locked.svg
            </svg>
          </div>
          <div
            className="card-body"
          >
            <h3
              className="card-title"
            >
              Security
            </h3>
            <p
              className="card-text"
            >
              A deep commitment to protecting generations of families.
            </p>
          </div>
        </div>
        <div
          className="card bg-primary border-0"
        >
          <div
            className="svg-container d-block mx-auto align-bottom"
          >
            <svg>
              handshake.svg
            </svg>
          </div>
          <div
            className="card-body"
          >
            <h3
              className="card-title"
            >
              Trust
            </h3>
            <p
              className="card-text"
            >
              150 years of strength and stability by your side.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
